test/solidity/Facets/ChainflipFacet.t.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.17;
+
+import { LibAllowList, TestBaseFacet, console, ERC20 } from "../utils/TestBaseFacet.sol";
+import { ChainflipFacet } from "lifi/Facets/ChainflipFacet.sol";
+import { LibAsset } from "lifi/Libraries/LibAsset.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+
+using stdJson for string;
+
+// Stub ChainflipFacet Contract
+contract TestChainflipFacet is ChainflipFacet {
+    constructor(address _chainflipVault) ChainflipFacet(_chainflipVault) {}
+
+    function addDex(address _dex) external {
+        LibAllowList.addAllowedContract(_dex);
+    }
+
+    function setFunctionApprovalBySignature(bytes4 _signature) external {
+        LibAllowList.addAllowedSelector(_signature);
+    }
+}
+
+contract ChainflipFacetTest is TestBaseFacet {
+    ChainflipFacet.ChainflipData internal validChainflipData;
+    TestChainflipFacet internal chainflipFacet;
+    address internal CHAINFLIP_VAULT;
+
+    uint256 internal constant CHAIN_ID_ETHEREUM = 1;
+    uint256 internal constant CHAIN_ID_ARBITRUM = 42161;
+    uint256 internal constant CHAIN_ID_SOLANA = 1151111081099710;
+    uint256 internal constant CHAIN_ID_BITCOIN = 20000000000001;
+
+    function setUp() public {
+        customBlockNumberForForking = 18277082;
+        initTestBase();
+
+        // Read chainflip vault address from config
+        string memory path = string.concat(
+            vm.projectRoot(),
+            "/config/chainflip.json"
+        );
+        string memory json = vm.readFile(path);
+        CHAINFLIP_VAULT = json.readAddress(".mainnet.chainflipVault");
```

Comment:
I like this...better than hardcoding. We could have a helper function for this where we just pass the name of the config file and the key, very similar to the one we use in our deploy scripts